department-of-veterans-affairs/abd-vro
# VRO Console: connect to Redis

## What was the problem?


As a follow-on to #531, we'd like examine contents of Redis in prod.

Associated tickets or Slack threads:

- #562
- #563 

## How does this fix it?


Adds Redis connection to the VRO Console.

## How to test this PR
- `export COMPOSE_PROFILES=debug` -- the console container only starts if the docker-compose `debug` profile is enabled.
- Start VRO. Note the `vro-console-1` container; check out the logs: `docker logs vro-console-1`.
- Generate a pdf using Swagger so that the pdf is added to Redis
- Attach to the container: `docker attach vro-console-1`
- On the `groovy:000>` prompt, try the following:
```groovy
groovy:000> redis.keys "*"
===> [1234_type, 1234]

// "1234" is the claim id
groovy:000> redis.get "1234_type"
===> hypertension

groovy:000> redis.get "1234"
// Shows the base64-encoded PDF contents

// Using RedisTemplate redisT
groovy:000> ops=redisT.opsForValue()
groovy:000> ops.get "1234_type"
===> hypertension
groovy:000> ops.get "1234"
// Shows the base64-encoded PDF contents
```
